### PR TITLE
Remove the browserid login code which was never used

### PR DESCRIPTION
**Pull request by magopian**
This was motivated by https://www.djangoproject.com/weblog/2015/aug/18/security-releases/ and an email from Michael Kelly:

```
FYI any sites using django-browserid are vulnerable, as we expose the logout view without a login_required check. Ya'll should upgrade if you're using it. :D

We considered issuing a django-browserid update to add that in, but if you're going to update to the latest django-browserid, you might as well upgrade Django as well.

If that's still a problem for your site, let me know why and we'll see what we can do.

Thanks!
- Mike Kelly 
```

### CODE REVIEW

File: requirements/prod.txt [hunk 1]
```diff
@@ -77,7 +75,6 @@ pytz==2013.8
 raven==4.2.3
 rdflib==3.0.0
 recaptcha-client==1.0.6
-receipts==0.2.9
```
Comment (magopian) [inline requirements/prod.txt]: Not related to the bug, but I believe it uses browserid. Also, we don't use it on AMO, it's a leftover from the marketplace before the guillotine.

File: requirements/prod.txt [hunk 2]
```diff
@@ -67,7 +66,6 @@ pyhs2==0.4.1
 polib==1.0.3
 protobuf==2.5.0
 pyasn1==0.1.7
-PyBrowserID==0.6
```
Comment (magopian) [inline requirements/prod.txt]: Not related to django-browserid, but while I was at it...

File: migrations/826-user-drop-source.sql [hunk 3]
```diff
@@ -0,0 +1 @@
+ALTER TABLE users DROP COLUMN `source`;
```
Comment (magopian) [inline migrations/826-user-drop-source.sql]: There are quite a few users that have the `source` coming from from marketplace:
```
mysql> select source, count(*) from users group by source;
+--------+----------+
| source | count(*) |
+--------+----------+
|      0 |  4500760 |
|      1 |    20194 |
|      2 |    89662 |
+--------+----------+
3 rows in set (2,19 sec)
```

For reference, the source values: https://github.com/mozilla/olympia/pull/669/files#diff-0383f36a36f392e46d9070ecdc5184a3L542
- 0: unknown source (all of the AMO users I believe)
- 1: users from the marketplace (without browserid)
- 2: users from the marketplace (with browserid)
- 3: users from amo (with browserid): there's none of those

**Comment by magopian** [comment]
Most of the changes are following a dumb search for `browserid` and removing everything that seemed related.

**Comment by kmaglione** [comment]
It was briefly used, on dev, but definitely a welcome cleanup.

File: migrations/826-user-drop-source.sql (hunk 3 above)
Comment (kmaglione) [inline migrations/826-user-drop-source.sql]: Maybe... but I wouldn't rush into it. Some of them might still be using their AMO accounts.

Comment (kmaglione) [inline migrations/826-user-drop-source.sql]:
```
MySQL [addons_mozilla_org]> select source, max(last_login_attempt) from users group by source order by source;^A
+--------+-------------------------+
| source | max(last_login_attempt) |
+--------+-------------------------+
|      0 | 2015-08-19 13:37:11     |
|      1 | 2015-08-19 06:27:05     |
|      2 | 2015-08-19 13:10:36     |
+--------+-------------------------+
```

Maybe we should just keep the column for now and decide how to deal with it later.

**Comment by kmaglione** [comment]
Very nice. I'd suggest leaving the `source` column in the database, and just deleting the index and model field for now, but I won't insist on it. r+

File: migrations/826-user-drop-source.sql (hunk 3 above)
Comment (magopian) [inline migrations/826-user-drop-source.sql]: That sure shows us that we shouldn't remove the users with source != 0, thanks!

I'm not sure if it's a good idea to have a "dangling" column like that, with no idea what it's for, no link with the model... what do you think @kmaglione ?

Comment (kmaglione) [inline migrations/826-user-drop-source.sql]: Yeah, I'm torn between not liking having a dangling column and not liking throwing away data that might be useful. But the more I think about it, the more I think it doesn't really matter, so let's just drop it.

**Pull request merged**